pkg/gatewayserver/io/ws/lbslns/upstream.go

```diff
@@ -581,20 +636,16 @@ func (f *lbsLNS) HandleUp(ctx context.Context, raw []byte, ids ttnpb.GatewayIden
 			ID: int32(txConf.XTime >> 48),
 		}
 		session.DataMu.Unlock()
-		recordRTT := true
-		refTime := getTimeFromFloat64(txConf.RefTime)
-		delta := receivedAt.Sub(refTime)
-		if delta > f.maxRoundTripDelay {
-			logger.WithFields(log.Fields(
-				"delta", delta,
-				"ref_time", refTime,
-				"received_at", receivedAt,
-			)).Warn("Gateway reported reftime greater than the valid maximum. Skip RTT measurement")
-			recordRTT = false
+		recordTime(0.0, txConf.XTime)
+
+	case TypeUpstreamTimeSync:
+		var req TimeSyncRequest
+		if err := json.Unmarshal(raw, &req); err != nil {
+			return nil, err
 		}
-		recordTime(recordRTT, txConf.RefTime, txConf.XTime, receivedAt)
+		return json.Marshal(req.Response(receivedAt))
```

Comment:
This message has it's own marshaler so you can use that right?

Reply:
Not a huge fan of calling the marshaller methods directly, but I can change this.